### PR TITLE
feat: React native datafile manager that supports Datafile caching.

### PR DESCRIPTION
**Pull request by zashraf1985**
## Summary
This overrides existing `HttpPollingDatafileManager` to support Datafile Caching for React Native. Browser and Node implementations can also override the same methods to support datafile caching.

## Test plan
Unit Tests Pending

### CODE REVIEW

File: packages/datafile-manager/__test__/httpPollingDatafileManager.spec.ts [hunk 1]
```diff
@@ -106,10 +100,6 @@ describe('httpPollingDatafileManager', () => {
       manager.start()
       expect(manager.responsePromises.length).toBe(1)
       await manager.responsePromises[0]
-      expect(updateFn).toBeCalledTimes(1)
-      expect(updateFn).toBeCalledWith({
-        datafile: { foo: 'bar' }
-      })
```
Comment (mjc1283) [inline packages/datafile-manager/__test__/httpPollingDatafileManager.spec.ts]: Why is this deleted?

Comment (mjc1283) [reply packages/datafile-manager/__test__/httpPollingDatafileManager.spec.ts]: Ok, I see why this behavior changed. We are only emitting update events after ready is resolved. Deleting this is OK.

File: packages/datafile-manager/src/httpPollingDatafileManager.ts [hunk 2]
```diff
@@ -314,4 +345,15 @@ export default abstract class HttpPollingDatafileManager implements DatafileMana
       logger.debug('Saved last modified header value from response: %s', this.lastResponseLastModified)
     }
   }
+
+  setDatafileFromCacheIfAvailable(): void {
+    this.cache.get(this.cacheKey)
+      .then(datafile => {
+        if (!this.isReadyPromiseSettled && datafile) {
```
Comment (mjc1283) [inline packages/datafile-manager/src/httpPollingDatafileManager.ts]: The datafile manager can be stopped at any time. When it's stopped, we want to cancel all pending timers, abort requests, and cut short any already-scheduled promise reactions. So let's add a check for `this.isStarted` in here and return early if it's false (just like other promise reactions in this class).

File: packages/datafile-manager/__test__/httpPollingDatafileManager.spec.ts [hunk 3]
```diff
@@ -634,4 +644,91 @@ describe('httpPollingDatafileManager', () => {
...
+      manager.queuedResponses.push(new Error('Connection Error'))
+      const updateFn = jest.fn()
+      manager.on('update', updateFn)
+      manager.start()
+      await manager.onReady()
+      expect(manager.get()).toEqual({name: 'keyThatExists'})
+      await advanceTimersByTime(50)
+      expect(manager.get()).toEqual({name: 'keyThatExists'})
+      expect(updateFn).toBeCalledTimes(0)
+      updateFn.mockReset()
```
Comment (mjc1283) [inline packages/datafile-manager/__test__/httpPollingDatafileManager.spec.ts]: Does calling `mockReset` matter? `updateFn` is just thrown away.

File: packages/datafile-manager/src/httpPollingDatafileManager.ts [hunk 4]
```diff
@@ -197,6 +226,8 @@ export default abstract class HttpPollingDatafileManager implements DatafileMana
     if (datafile !== null) {
       logger.info('Updating datafile from response')
       this.currentDatafile = datafile
+      logger.debug('Adding Datafile to Cache')
```
Comment (mjc1283) [inline packages/datafile-manager/src/httpPollingDatafileManager.ts]: Let's remove this log message. It will be confusing for users to see this with the no-op cache.

File: packages/datafile-manager/__test__/httpPollingDatafileManager.spec.ts (hunk 3 above)
Comment (zashraf1985) [reply packages/datafile-manager/__test__/httpPollingDatafileManager.spec.ts]: you are right. I naively copied without thinking. The places where i copied it from was verifying it twice hence resetting after first use. Pifalls of copy/pasting :)
